### PR TITLE
Match func_8006A1C4 as handle_save_data_and_read_controller

### PR DESCRIPTION
**Pull request by Ryan-Myers**
This one only matches a single function, but it does a LOT of documenting because of it, and I figured it would be better to push it now rather than let it grow.

### CODE REVIEW

File: src/menu.h [hunk 1]
```diff
@@ -818,7 +818,7 @@ s32 get_filtered_cheats(void);
 s32 get_number_of_active_players(void);
 s32 get_active_player_count(void);
 s32 get_multiplayer_racer_count(void);
-Settings** func_8009C490(void);
+Settings** get_all_save_files_ptr(void);
```
Comment (FazanaJ) [inline src/menu.h:821]:
```suggestion
Settings **get_all_save_files_ptr(void);
```

File: src/game.c [hunk 2]
```diff
@@ -1818,7 +1822,7 @@ void func_8006E770(Settings *settings, s32 arg1) {
 
     get_number_of_levels_and_worlds(&numLevels, &numWorlds);
     temp_v0 = (u16 *)get_misc_asset(MISC_ASSET_UNK17);
-    for (i = 0; i < 3; i++) { // 3 = number of save files?
+    for (i = 0; i < NUMBER_OF_SAVE_FILES; i++) { // 3 = number of save files?
```
Comment (DavidSM64) [inline src/game.c:1825]: Comment needs to be removed

**Review by DavidSM64 — COMMENTED** [review]
Other than this one little comment, lgtm

**Review by DavidSM64 — APPROVED** [review]
Other than this one little comment, LGTM

**Pull request merged**